src/wallet/coinselection.cpp

```diff
@@ -0,0 +1,370 @@
+  {
+    typedef SpendableTxos::const_iterator T;
+    bool operator() (const T& x, const T& y) const {return x->first < y->first;}
+  };
+
+typedef std::set<SpendableTxos::iterator,TxoIterLess> TxoItVec;
+typedef std::pair<CAmount, TxoItVec > TxoGroup;  // A set of coins and how much they sum to.
+#endif
+
+// This holds a sorted map of coin amounts and the set of Txos required to get to that amount
+typedef std::multimap<CAmount, TxoGroup> TxoGroupMap;
+
+// How close do we want to get to targetValue (at first).  The longer it takes the more we add to this value.
+CAmount reasonableExcess(CAmount targetValue) { return targetValue / 1024; }
+// Create a group of Txos from a Txo index and another group.
+TxoGroup makeGroup(SpendableTxos::iterator i, const TxoGroup *prev = nullptr)
+{
+    TxoGroup ret;
+    if (prev != nullptr)
+        ret = *prev;
+    else
+        ret.first = 0;
+    // Don't add an element that already exists  -- allow this, bad solutions eliminated later
+    // assert(ret.second.find(i) == ret.second.end());
+    ret.first += i->first; // update the amount
+    COutput outp = i->second;
+    // verify that the output value is consistent -- TODO: Check pcoin and make sure it doesn't get deleted -- located
+    // in mapWallet
+    assert(outp.tx->vout[outp.i].nValue == i->first);
+    ret.second.insert(i); // add this txo to the set
+    return ret;
+}
+
+// Take a group that is not a solution and find a bunch of solutions by appending additional TXOs onto it.
+// This is a recursive function that is limited by MAX_TXOs and MAX_ELECTIVE_TXOS
+bool extend(const CAmount targetValue,
+    /*const*/ SpendableTxos &available,
+    TxoGroup grp,
+    TxoGroupMap &solutions,
+    int depth)
+{
+    if (depth >= MAX_TXOS)
+        return false; // unambiguously break the recursion
+    SpendableTxos::iterator aEnd = available.end(); // For speed, get these once
+    SpendableTxos::iterator aBeg = available.begin();
+    bool ret = false;
+    SpendableTxos::iterator small;
+    small = available.lower_bound(targetValue - grp.first); // Find a value at or above what we need
+    if (1)
+    {
+        SpendableTxos::iterator i = small;
+        // while not at the end and there's too little value or we already used this output then keep going.
+        // while((i != aEnd) && ((i->first + grp.first < targetValue) || (grp.second.find(i) != grp.second.end()))  )
+        // allow dups and eliminate later for performance
+        while ((i != aEnd) && (i->first + grp.first < targetValue)) // iterate to make sure it is actually big enough.
+        {
+            ++i;
+        }
+
+        if (i != aEnd) // Ok we have a solution so add it.
+        {
+            TxoGroup g = makeGroup(i, &grp);
+            solutions.insert(TxoGroupMap::value_type(g.first, g));
+            // LogPrint("wallet","CoinSelection solution found: excess %ld, txos: %d\n", g.first - targetValue,
+            // g.second.size());
+            ret = true;
```

Comment:
do you still need this commented logpring?

Reply:
I guess not.  I will remove to get this moving.  But as I was saying in a different PR, we need to discuss this policy.  Reasons to leave it in:

The commented logprint acts as a comment describing the code.  And if uncommented it provides great detail about the operation of the algorithm, but its so log-spammy that you'd never want to use it unless you were specifically debugging or enhancing this area of the code.  But if you ARE doing so, its a lot easier to uncomment this log than to recreate it.  So in sum, I think that there is a role in the code for commented-out log statements...